### ISSUE TITLE
Allow custom naming for volumes in version 2 files

### ISSUE THREAD

**Issue by mrapczynski** [docker/compose#3695]
Take for example the following Compose YAML:

``` yaml
version: '2'

volumes:
    docker.plugin-test:
        driver: ceph-rbd

services:
    plugin-test:
        environment:
            - constraint:environment==test
            - reschedule:on-node-failure
        image: busybox
        volume_driver: ceph-rbd
        volumes:
            - docker.plugin-test:/plugin-test
```

The problem I have is Compose decides to prefix the project directory name onto the volume name. So if I have a directory of YML files named `compose`, then instead of setting up my volume with the **declared** name `docker.plugin-test`, it tries to mount a **non-existent volume** named `compose_docker.plugin-test`. Overriding the project name to be blank such as with `-p ""` does not seem to work.

This is proving to be really annoying because it is inconsistent with the other ways volumes are mounted onto a container through a plugin. `docker volume create ...` allows me to specify the exact name I want. Compose v1 or `docker run --volume-drive=ceph-rbd -v docker.plugin-test:/data` also respect the name I provide.

I don't want Compose or my plugin to always be expected to create and manage an external volume. Most often I create and format the block device outside of Docker for testing before mounting it onto a container. So I expect the name I already chose to be honored.

So as to not break existing behavior for developers who may already be used to this, can we allow volumes with custom drivers to override the name that Compose would choose? For example:

``` yaml
version: '2'

volumes:
    docker.plugin-test:
        driver: ceph-rbd
        name: docker.plugin-test

services:
    plugin-test:
        environment:
            - constraint:environment==test
            - reschedule:on-node-failure
        image: busybox
        volume_driver: ceph-rbd
        volumes:
            - docker.plugin-test:/plugin-test
```

**Comment by aanand** [comment]
I'd argue that this behaviour is fully consistent with how Compose names containers and networks, which is necessarily different from how the lower-level Engine commands behave.

It sounds like you want the [`external` option](https://docs.docker.com/compose/compose-file/#/external):

```yaml
volumes:
  docker.plugin-test:
    external: true
```

You can use a different internal name for the volume, if you prefer:

```yaml
volumes:
  my-volume:
    external:
      name: docker.plugin-test

services:
  plugin-test:
    volumes:
      - my-volume:/plugin-test
```

**Comment by mrapczynski** [comment]
@aanand Alright, tried that and Docker complains. Not surprising because now a volume plugin/driver is no longer specified.

```ERROR: Volume docker.plugin-test declared as external, but could not be found. Please create the volume manually using `docker volume create --name=docker.plugin-test` and try again.```

My RBD volumes 99.9% of the time will already exist. The expectation is that the Docker engine can just request the plugin to mount the named block device. I get what v2 is trying to accomplish by allowing the coordinating of multiple services, but if I don't want a project name forced, then I should be able to get around that.

Part of why the volume name is so important is the portion before the ```.``` specifies the storage pool where the image resides - it's essential part of mounting the device.

**Comment by aanand** [comment]
OK, so perhaps we should change [this error](https://github.com/docker/compose/blob/e5529a8/compose/volume.py#L93-L101) to a warning. That way, if the volume plugin is capable of creating the volume on demand, then it should work. You'll still need to specify `volume_driver` for each service that uses it.

Would that fix your issue?

**Comment by mrapczynski** [comment]
@aanand OK that makes a lot more sense now because specifying the driver on the service is very clear what the runtime expectation is

**Comment by aanand** [comment]
@dnephin WDYT?

**Comment by dnephin** [comment]
Warning instead of error sounds good to me

**Comment by renier** [comment]
I would prefer being able to specify the name being used to create the volume, rather than pre-creating it externally, just like `container_name` now allows you to specify the specific container name you want. Maybe a `volume_name` is in order?

**Comment by shin-** [comment]
Fixed by #5096
